Add cross run config in analyzer

Fredy-Z opened the pull request:
<!--
Request Prow to automatically lint any go code in this PR:

/lint
-->

## Proposed Changes
* `Mako` has supported `CrossRunConfig` in `ThresholdAnalyzer`, update to use it. Now the analyzer will only fail if there are 10 continuous runs that cross the preset threshold.

**Release Note**

<!-- Enter your extended release note in the below block. If the PR requires
additional action from users switching to the new release, include the string
"action required". If no release note is required, write "NONE". -->

```release-note
NONE
```

/cc @srinivashegde86 
/cc @vagababov 
/cc @mattmoor

Fredy-Z commented:
/retest

srinivashegde86 commented on test/performance/dataplane-probe/sla.go:
```diff
@@ -41,6 +41,12 @@ var (
 				ValueKey:            proto.String("kd"),
 			},
 		}},
+		CrossRunConfig: &tpb.CrossRunConfig{
```
too many duplicates. Can we have a common var and use that? Maybe even move it to mako package so that all tests can use this default?

Fredy-Z replied:
Moved to using a separate function to construct the config. Added a placeholder for `tags` in case we want to search for historic runs that has the same tags.

vagababov commented on test/performance/mako/analyzer.go:
```diff
@@ -0,0 +1,35 @@
+func NewCrossRunConfig(runCount int, tags ...string) *tpb.CrossRunConfig {
+	return &tpb.CrossRunConfig{
+		RunInfoQueryList: []*mpb.RunInfoQuery{{
+			Limit: proto.Int32(10),
```
should we use `runCount` ?

Fredy-Z replied:
Ah sorry I missed it when I moved the code, replaced with `runCount`.

mattmoor commented on test/performance/mako/analyzer.go:
```diff
@@ -0,0 +1,35 @@
+
+// NewCrossRunConfig returns a config that can be used in ThresholdAnalyzer.
+// By using it, the Analyzer will only fail if there are xx continuous runs that cross the threshold.
+func NewCrossRunConfig(runCount int32, tags ...string) *tpb.CrossRunConfig {
```
we aren't passing in any tags from our `sla.go` files...

Fredy-Z replied:
I'm thinking about leaving it as a placeholder in case we want to add tags in the future, or do you want me to remove it now?

mattmoor replied:
My point is that if this is what's used to establish the window then I think it's incorrect for at least the load-test, which uses tagging to distinguish multiple data points.

Fredy-Z replied:
Fixed.
Initially I was thinking `Mako` should be able to determine which tag(s) to use for the query, but since we are using a few other tags, like commitID and kube version, it makes sense to let us specify the tag(s).

Fredy-Z commented:
> /lgtm
> /approve
> /hold
> 
> This needs a new sidecar, does it not?

Yes, it's in gcr.io/test-project-chizhg/mako-microservice:latest, you should be able to `crane` it to `mattmoor-public` :-) ?

mattmoor commented:
Not if your project is private.  Ping me tomorrow and let's chat about it.

Fredy-Z commented:
> Not if your project is private. Ping me tomorrow and let's chat about it.

As we discussed, I've moved the image to `knative-performance`, and replaced in yaml files.

Fredy-Z commented:
/hold cancel

Fredy-Z commented:
/retest

Fredy-Z commented:
/retest

Fredy-Z commented:
(╯°□°）╯︵ ┻━┻
/retest
